### PR TITLE
`Development`: Refactor optional usage

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
This is a followup for #6755. Additionally, to just performing search and replace, this PR refactors code smells regarding optional usage.

### Steps for Testing
code review

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Replacements look good to me 👍

File: src/main/java/de/tum/in/www1/artemis/service/hestia/TestwiseCoverageService.java
Comment (jakubriegel) [inline src/main/java/de/tum/in/www1/artemis/service/hestia/TestwiseCoverageService.java:298]: We could use `coverageReportRepository::findCoverageReportByIdWithEagerFileReportsAndEntries()` directly because if `optionalLazyReport` is not empty then that report exists and classes that use this method (`BehavioralTestCaseService`, `CoverageReportResource`) have handlers for empty optional anyway.


Btw, this would be more readable with an intermediate `map()`:
```java
optionalLazyReport.map(CoverageReport::getId)
    .map(CoverageReportRepository::findCoverageReportByIdWithEagerFileReportsAndEntries)
```

**Review by jakubriegel — CHANGES_REQUESTED** [review]

**Review by jakubriegel — DISMISSED** [review]

File: src/main/java/de/tum/in/www1/artemis/service/hestia/TestwiseCoverageService.java
Comment (jakubriegel) [reply src/main/java/de/tum/in/www1/artemis/service/hestia/TestwiseCoverageService.java]: Resolved by adding dedicated database query

**Review by MichaelOwenDyer — APPROVED** [review]
LGTM 👍🏽

**Review by simon-hng — DISMISSED** [review]

**Review by b-fein — APPROVED** [review]

**Review by b-fein — APPROVED** [review]

**Review by b-fein — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Reapprove after merges

**Review by jakubriegel — APPROVED** [review]
Code LGTM after merge 👍🏻 

Teamscale complains about some minor issues, but I think they can be marked as tolerated

**Review by simon-hng — APPROVED** [review]

**Pull request merged**
